### PR TITLE
WT-5247 Resizing modify operation log full values.

### PR DESCRIPTION
**Pull request by michaelcahill**
When modify operations resize part of a value, subsequent bytes are shuffled around and the operation is no longer idempotent.  Since WiredTiger's recovery can reapply operations that overlap with a checkpoint, it is not safe to log resizing modify operations.  Instead, log full update operations for that case.

### CODE REVIEW

**Comment by agorrod** [comment]
@tetsuo-cpp and @keithbostic please review this change. It's intent is to fix an issue earlier versions of MongoDB run into, where they do modify operations on logged tables. It's not (very) relevant for current versions of MongoDB, since they don't generally do modify operations on logged tables.
It's not safe to do a modify, because the generated log message isn't idempotent at the moment (i.e: if it get's applied twice you get different results than if it's only applied once).

File: src/support/modify.c [hunk 1]
```diff
@@ -38,6 +38,33 @@
...
+    size_t tmp;
+    const size_t *p;
+    int nentries;
+
+    /* Get the number of modify entries. */
+    p = modify;
+    memcpy(&tmp, p++, sizeof(size_t));
+    nentries = (int)tmp;
+
+    WT_MODIFY_FOREACH_BEGIN(mod, p, nentries, 0)
```
Comment (tetsuo-cpp) [inline src/support/modify.c]: Nit: Not introduced by your change but could you please add `WT_MODIFY_FOREACH_BEGIN` to the `.clang-format` file as a loop macro? I must have missed this one.

File: src/support/modify.c [hunk 2]
```diff
@@ -38,6 +38,33 @@
...
+    {
+        if (mod.size != mod.data.size)
```
Comment (tetsuo-cpp) [inline src/support/modify.c]: I assume it's correct but even after looking at the struct definition, I don't what it means for the modify size to be different from its data size. A brief comment would be nice for us newbies.

File: src/support/modify.c (hunk 1 above)
Comment (michaelcahill) [reply src/support/modify.c]: Done.  Also sorted the list while I was in the area.

File: src/support/modify.c (hunk 2 above)
Comment (michaelcahill) [reply src/support/modify.c]: I added a comment: let me know if that helps.

**Comment by tetsuo-cpp** [comment]
@michaelcahill Something went wrong with your last commit. I don't see the changes you described but I see the `clang-format` binary + some SWIG generated file.

**Comment by michaelcahill** [comment]
Sigh, thanks @tetsuo-cpp, that was my first attempt to use Visual Code end-to-end.  I'll fix manually and push...

File: src/support/modify.c (hunk 2 above)
Comment (tetsuo-cpp) [reply src/support/modify.c]: I get it now. Thanks.

**Pull request merged**